connector/routingconnector/logs_test.go

```diff
@@ -0,0 +1,329 @@
+		rl.ScopeLogs().AppendEmpty().LogRecords().AppendEmpty()
+
+		require.NoError(t, conn.ConsumeLogs(context.Background(), l))
+
+		assert.Len(t, defaultSink.AllLogs(), 1)
+		assert.Len(t, sink0.AllLogs(), 1)
+		assert.Len(t, sink1.AllLogs(), 1)
+
+		assert.Equal(t, sink0.AllLogs(), sink1.AllLogs())
+
+		rlog := defaultSink.AllLogs()[0].ResourceLogs().At(0)
+		attr, ok := rlog.Resource().Attributes().Get("X-Tenant")
+		assert.True(t, ok, "routing attribute must exists")
+		assert.Equal(t, attr.AsString(), "something-else")
+	})
+
+	t.Run("logs matched by one expression, multiple pipelines", func(t *testing.T) {
+		resetSinks()
+
+		l := plog.NewLogs()
+
+		rl := l.ResourceLogs().AppendEmpty()
+		rl.Resource().Attributes().PutStr("X-Tenant", "ecorp")
+		rl.ScopeLogs().AppendEmpty().LogRecords().AppendEmpty()
+
+		require.NoError(t, conn.ConsumeLogs(context.Background(), l))
+
+		assert.Len(t, defaultSink.AllLogs(), 1)
+		assert.Len(t, sink0.AllLogs(), 1)
+		assert.Len(t, sink1.AllLogs(), 0)
+
+		assert.Equal(t, defaultSink.AllLogs()[0].LogRecordCount(), 1)
+		assert.Equal(t, sink0.AllLogs()[0].LogRecordCount(), 1)
+		assert.Equal(t, defaultSink.AllLogs(), sink0.AllLogs())
+	})
+}
+
+func TestLogsResourceAttributeDroppedByOTTL(t *testing.T) {
+	logsDefault := component.NewIDWithName(component.DataTypeLogs, "default")
+	logsOther := component.NewIDWithName(component.DataTypeLogs, "other")
+
+	cfg := &Config{
+		DefaultPipelines: []component.ID{logsDefault},
+		Table: []RoutingTableItem{
+			{
+				Statement: `delete_key(attributes, "X-Tenant") where attributes["X-Tenant"] == "acme"`,
+				Pipelines: []component.ID{logsOther},
+			},
+		},
+	}
+
+	sink0 := &consumertest.LogsSink{}
+	sink1 := &consumertest.LogsSink{}
+
+	consumer := fanoutconsumer.NewLogsRouter(
+		map[component.ID]consumer.Logs{
+			logsDefault: sink0,
+			logsOther:   sink1,
+		})
+
+	factory := NewFactory()
+	conn, err := factory.CreateLogsToLogs(
+		context.Background(),
+		connectortest.NewNopCreateSettings(),
+		cfg,
+		consumer,
+	)
+
+	require.NoError(t, err)
+	require.NotNil(t, conn)
+	require.NoError(t, conn.Start(context.Background(), componenttest.NewNopHost()))
+	defer func() {
+		assert.NoError(t, conn.Shutdown(context.Background()))
+	}()
+
+	l := plog.NewLogs()
+	rm := l.ResourceLogs().AppendEmpty()
+	rm.Resource().Attributes().PutStr("X-Tenant", "acme")
+	rm.Resource().Attributes().PutStr("attr", "acme")
+
+	assert.NoError(t, conn.ConsumeLogs(context.Background(), l))
+	logs := sink1.AllLogs()
+	require.Len(t, logs, 1, "log should be routed to non-default exporter")
+	require.Equal(t, 1, logs[0].ResourceLogs().Len())
+	attrs := logs[0].ResourceLogs().At(0).Resource().Attributes()
+	_, ok := attrs.Get("X-Tenant")
+	assert.False(t, ok, "routing attribute should have been dropped")
+	v, ok := attrs.Get("attr")
+	assert.True(t, ok, "non routing attributes shouldn't be dropped")
+	assert.Equal(t, "acme", v.Str())
+	require.Len(t, sink0.AllLogs(), 0,
```

Comment:
Shouldn't this be an assert instead? I would even argue that all requires in this part should be asserts, as this is in the verification phase of the test, but I see why the others are `require` (just not this one)